File: 2020/06/main.js
import { readFileSync } from "fs";

const input = readFileSync("2020/06/input.txt")
	.toString()
	.trim()
	.split("\n\n")
	.map((group) => group.split("\n"));

let answer1 = 0;
let answer2 = 0;
const aCharCode = "a".charCodeAt(0);
for (const group of input) {
	const yesCount = Array(26).fill(0);
	for (const person of group) {
		for (const c of person) {
			yesCount[c.charCodeAt(0) - aCharCode] += 1;
		}
	}
	for (const count of yesCount) {
		if (count > 0) answer1++;
		if (count === group.length) answer2++;
	}
}

console.log("part 1: " + answer1);
console.log("part 2: " + answer2);

File: 2020/06/main.zig
const std = @import("std");
const print = std.debug.print;

const input = @embedFile("input.txt");

pub fn main() !void {
    var answer1: u32 = 0;
    var answer2: u32 = 0;
    var groups = std.mem.split(input, "\n\n");
    while (groups.next()) |group| {
        var yes_answers = [_]u32{0} ** 26;
        var group_len: u32 = 0;
        var people = std.mem.tokenize(group, "\n");
        while (people.next()) |person| {
            for (person) |c| {
                yes_answers[c - 'a'] += 1;
            }
            group_len += 1;
        }
        for (yes_answers) |count| {
            if (count > 0) answer1 += 1;
            if (count == group_len) answer2 += 1;
        }
    }
    print("part 1: {}\n", .{answer1});
    print("part 2: {}\n", .{answer2});
}
